server.go

```diff
@@ -713,8 +716,9 @@ func (s *baseServer) handleInitialImpl(p receivedPacket, hdr *wire.Header) error
 		delete(s.zeroRTTQueues, hdr.DestConnectionID)
 	}
 
-	go conn.run()
+	s.handshakingCount.Add(1)
 	go func() {
```

Comment:
This goroutine can push a connection on the conn queue right when the server is closing. Is that a problem?

Reply:
Isn't there a race here? 
```Accept ...
	select {
	case <-ctx.Done():
		return nil, ctx.Err()
	case conn := <-s.connQueue:
		return conn, nil
	case <-s.errorChan:
		return nil, s.closeErr
	}
```

Reply:
connQueue has a connection and errorChan is closed.

Reply:
That is true. It's independent from this PR though. I'll open an issue.

Reply:
#4744